### DISCUSSION TITLE
Option to hide cursor in overlapping popup

### DISCUSSION THREAD

**Discussion by habamax** [vim/vim#14227] [Ideas]
I remember Bram was strict about having cursor always visible if popup overlaps it.

But it was probably before ppl figured out how to fake input boxes using popups. And if you have it faked, there are 2 visible cursors that is less than ideal.

Would it make sense to have a popup option to hide cursor if it is overlapped?

![image](https://github.com/vim/vim/assets/234774/c7726234-e887-41e9-9de1-d6de4cd8cd48)

**Comment by Shougo** [comment]
https://github.com/vim/vim/pull/14436 can fix the problem.

**Comment by Shougo** [comment]
You can use `set t_ve=` in CUI Vim.  It does not work for GVim.  But in GVim `Cursor` highlight works instead.

**Reply by habamax** [reply]
yes, indeed. thx!

**Reply by Shane-XB-Qian** [reply]
i am curious where that second cursor came from, maybe that's not a pure vim popup,
if i guess correctly, then that may originally can hide the cursor by raw esc code already.

**Reply by habamax** [reply]
That is a fake cursor in a pure vim popup.

**Comment by habamax** [comment]
I have workaround-ed it with 

```
    # hide cursor
    set t_ve=
    var gui_cursor = hlget("Cursor")
    hlset([{name: 'Cursor', cleared: true}])
```

and

```
    # restore cursor
    set t_ve&
    if hlget("Cursor")[0]->get('cleared', false)
        hlset(gui_cursor)
    endif
```
